### ISSUE TITLE
Compare to the predecesor, not to the latest of the branch

### ISSUE THREAD

**Issue by mgrojo** [OpenNingia/vscode-clearcase#174]
I expected that the comparison from the Source Control panel is done with the predecessor, but it is done with the latest version on the branch.

Usually, you'd see the same changes, but if someone else is working on the same branch and makes a checkin, you see differences which you don't expect.

**Comment by fr43nk** [comment]
Hi,
it is a bit hard for me to understand:

```
      [main]
        |
       [1]
        |
       [2]
        |
       [3]
        |
       [4] -> coworkers checkout based on
        |
       [5] -> your checkout based on
```
    
So, if your coworker checks in, a new version is created and the diff basis changes? So, you need the version at the moment of your checkout?
It sounds quite confusing for me. I think this should never happen as this is what branches are for. I work here in a quite big software project and such a scenario never happens.

But, I plan an new feature to let the user choose the version for a diff of the current open file (not the scm view), so maybe that would help you.

Best regards,
Frank

**Comment by mgrojo** [comment]
Hi,

>  It sounds quite confusing for me. I think this should never happen as this is what branches are for. I work here in a quite big software project and such a scenario never happens.

In our case is also unusual, but not impossible, that two coworkers are working (or just merging) on the same branch. So an example of that scenario is:

```
      [main]
        |
       [0]
        |
       [2]---[work-branch]
        |                     |
       [3]                  [0] -> coworker's and my checkout are based on this
                              / \
 [other checkout]  [my checkout]
```
Up to this situation the differences are expected, they show my changes [my checkout] from the base of my checkout [0].

After check-in of coworker:
```
      [main]
        |
       [0]
        |
       [2]---[work-branch]
        |                     |
       [3]                  [0] -> coworker's and my checkout are based on this
                                | \
                              [1]  [my checkout]
```
Now the differences are unexpected, they show my changes and the (fake) removal of my coworker changes; [my checkout] compared to the branch LATEST [1].

My expectation is that this should always work the same as the `-predecessor` switch of `cleartool diff`. It's easy to get the predecessor in cleartool commands; as documented in fmt_ccase:

> %PSn
>
> Predecessor Short name: For a version, a short form of the predecessor version's version ID: branch-pathname/version-number. For other objects, the null string.
>
> %PVn
>
> Predecessor Version ID: For a version, the predecessor version's version ID; for other objects, the null string.

Of course, you can always call the ClearCase diff tool, which will always compare to the predecessor, but then, the integrated SCM view loses sense.

> But, I plan an new feature to let the user choose the version for a diff of the current open file (not the scm view), so maybe that would help you.

That's great; it's an important feature on its own and will also mitigate this issue. But it is the unexpected viewing of changes you haven't done what should be avoided, specially when you don't know yet that it is not showing changes to the predecessor, but to the branch latest.

**Comment by fr43nk** [comment]
Nice, thank you for the explanation. I wasn't aware of this. I will check what can be done to address this issue. Perhaps I create a new settings property like for the checkout or checkin command.

**Comment by fr43nk** [comment]
It is fixed with #175
